pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -182,6 +192,62 @@ func (r *NSTemplateSetReconciler) fetchNamespaces(nsTemplateSetName string) ([]c
 
 }
 
+// ensureClusterResources ensures that the cluster resources exists.
+// Returns `true, nil` if something was created or updated, `false, nil` if nothing changed, `true|false, err` if an
+// error occurred
+func (r *NSTemplateSetReconciler) ensureClusterResources(logger logr.Logger, nsTmplSet *toolchainv1alpha1.NSTemplateSet) (bool, error) {
+	logger.Info("ensuring cluster resources", "username", nsTmplSet.GetName(), "tier", nsTmplSet.Spec.TierName)
+	clusterResources := nsTmplSet.Spec.ClusterResources
+	// cluster-wide resources are optional. Skip if none was specified in the given NSTemplateSet
+	if clusterResources == nil {
+		return false, nil
+	}
+
+	tmpl, err := r.getTemplateContent(nsTmplSet.Spec.TierName, ClusterResources)
+	if err != nil {
+		return false, r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusClusteResourcesProvisionFailed, err, "failed to to retrieve template for the cluster resources")
+	}
+
+	tmplProcessor := template.NewProcessor(r.client, r.scheme)
+	params := map[string]string{"USERNAME": nsTmplSet.GetName()}
+	objs, err := tmplProcessor.Process(tmpl, params)
+	if err != nil {
+		return false, r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusClusteResourcesProvisionFailed, err, "failed to to retrieve template for the cluster resources")
+	}
+
+	for _, rawObj := range objs {
+		acc, err := meta.Accessor(rawObj.Object)
+		if err != nil {
+			return false, r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusClusteResourcesProvisionFailed, err, "invalid element in template for the cluster resources")
+		}
+
+		// set labels
+		labels := acc.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		labels[toolchainv1alpha1.OwnerLabelKey] = nsTmplSet.GetName()
+		labels[toolchainv1alpha1.TypeLabelKey] = ClusterResources
+		labels[toolchainv1alpha1.TierLabelKey] = nsTmplSet.Spec.TierName
+		acc.SetLabels(labels)
```

Comment:
There is another WIP PR to add the provider key in the controller rather than relay on the template manifests: https://github.com/codeready-toolchain/member-operator/pull/142/files#diff-cffea82f7fdc7427b8c9f0d0fdd55aedR309
So, just to make sure we don't miss this for the cluster resources let's add it here too:
```suggestion
		labels[toolchainv1alpha1.ProviderLabelKey] = toolchainv1alpha1.ProviderLabelValue
		acc.SetLabels(labels)
```